float16_t.hpp

```diff
@@ -560,7 +562,9 @@ namespace numeric
     {
         float16_t_private::float16 data_;
 
+#if __cpp_constexpr >= 201907L
         constexpr inline float16_t() noexcept = default;
```

Comment:
What happens if this #if clause is not verified? Is it still possible to construct a float16_t without any argument? I believe this should be added as an #else:

```
constexpr inline float16_t() noexcept : data_{ 0 } { }
```

Reply:
You are right, I missed that. But for the #else block, is it add
```
inline float16_t() noexcept = default;
```
better ? Since `__cpp_constexpr < 201907L`, which means float16_t_private::float16's trivial default initialization can't be called in constexpr function.